pvanalytics/quality/irradiance.py

```diff
@@ -592,3 +592,260 @@ def calculate_component_sum_series(solar_zenith,
+                                   0.01 * dni_extra * cosd(solar_zenith))
+
+    # Setting lower limit as undefined where solar_zenith is not available
+    lower_limit[solar_zenith.isna()] = np.nan
+
+    return lower_limit
+
+
+def check_poa_global_limits_pvlive(poa_global, solar_zenith, aoi,
+                                   dni_extra=1367.):
+    r"""Test for limits on POA global with pvlive algorithm.
+
+    Criteria from Section 6.1 of [1]_ are used to determine physically
+    plausible lower and upper bounds. Each value is tested and a value passes
+    if value > lower bound and value < upper bound. Also, steps with
+    change in magnitude of more than 1000 W/m2 are flagged. Lower bounds are
+    constant for all tests. Upper bounds are calculated as
+
+    .. math::
+        upper\_limit = 0.9 * dni\_extra * cos(aoi)^{1.2} + 300
+
+    Parameters
+    ----------
+    poa_global : Series
+        Global tilted irradiance [W/m^2]
+    solar_zenith : Series
+        Solar zenith angle [degrees]
+    aoi : Series
+        angle of incidence [degrees]
+    dni_extra : float, default 1367.
+        normal irradiance at the top of atmosphere [W/m^2]
+
+    Returns
+    -------
+    poa_global_limit_bool_flag : Series
+        True for each value that is physically possible.
+    poa_global_limit_int_flag : Series
+        Series of integers representing the flag numbers described in the
+        literature. [1]_
+
+    Notes
+    -----
+    The upper limit for `poa_global` is set to 0 when `solar_zenith` is greater
+    than 90 degrees. Missing values of `poa_global`, `solar_zenith`
+    and/or `aoi` will result in a `False` flag.  Also, [1]_ mentions that the
+    proposed limits are for silicon sensors with a tilt of 25°.
+
+    References
+    ----------
+    .. [1] Elke Lorenz et al., High resolution measurement network of global
+           horizontal and tilted solar irradiance in southern Germany with a
+           new quality control scheme, Solar Energy, Volume 231, 2022,
+           Pages 593-606, ISSN 0038-092X,
+           https://doi.org/10.1016/j.solener.2021.11.023.
+    """
+    # Finding the upper and lower limit
+    upper_limit = _upper_poa_global_limit_pvlive(aoi, solar_zenith, dni_extra)
+    lower_limit = _lower_limit_pvlive(solar_zenith, dni_extra)
+
+    # Initiating a poa_global_limit_int_flag series
+    poa_global_limit_int_flag = pd.Series(0, index=solar_zenith.index)
+
+    # Changing the poa_global_flag to 3 when poa_global is above upper
+    # limit or below lower limit
+    poa_global_limit_int_flag = poa_global_limit_int_flag.mask(
+        ((poa_global > upper_limit) |
+         (poa_global < lower_limit)),
+        3
+    )
+
+    # Changing the poa_global_flag to 3 when the step change in poa values is
+    # more than 1000 W/m2
+    poa_global_limit_int_flag = poa_global_limit_int_flag.mask(
+        poa_global.diff().abs() > 1000,
+        3
+    )
```

Comment:
I think this deserves it's own function. 

The step change limit is fairly common:

> For 1-min data Shafer et al. (2000) recommended a step change limit of 800 W/m2, although Espinar (2011) suggested 1000 W/m2 for GHI. Lorenz et al. (2022) also suggested using the 1000 W/m2 threshold for tilted irradiance measurements.

Reply:
It could admittedly be done in a separate pull request.

Reply:
Good point. I think one good reason to have the step change limit in here is to stay true to the list of checks the paper suggests.

Reply:
we could have a helper function that returns the step threshold, and pass it a kwarg 'pvlive' to return 1000.

Reply:
> However, I am completely fine with separating out the filter as a separate function ☺️.

As am I, let's do this and have a `step_limit` kwarg or something.

Reply:
> we could have a helper function that returns the step threshold, and pass it a kwarg 'pvlive' to return 1000.

Can somebody help me with this? After some reading, I think I understand kwargs, but have never written a function with them. Some more guidance on how the functions' design would be helpful.